modules/transactions.js

```diff
@@ -73,39 +73,90 @@ __private.attachApi = function () {
 
 __private.list = function (filter, cb) {
 	var sortFields = sql.sortFields;
-	var params = {}, where = [], owner = '';
+	var params = {},
+		where = [],
+		err,
+		owner = '',
+		allowedFieldsMap = {
```

Comment:
That should be performed by schema validation (and is), hovewer there are some parameters that schema allow to be empty, but I don't want them empty - so that kind of hack is used for transitional period. That should be fixed in schema, for example: not allow publicKey to be empty.